app/controllers/api/storage.php

```diff
@@ -421,7 +431,7 @@
 
         $maximumFileSize = $bucket->getAttribute('maximumFileSize', 0);
         if ($maximumFileSize > (int) App::getEnv('_APP_STORAGE_LIMIT', 0)) {
-            throw new Exception('Error bucket maximum file size is larger than _APP_STORAGE_LIMIT', 500, Exception::GENERAL_SERVER_ERROR);
+            throw new Exception('Error bucket maximum file size is larger than _APP_STORAGE_LIMIT', 400, Exception::STORAGE_INVALID_FILE_SIZE);
```

Comment:
Why are we using a different error code here ? Compared to `ATTRIBUTE_VALUE_INVALID ` in the previous cases ?

Reply:
One is client-side, and one is server-side endpoint. The client will get INVALID_FILE_SIZE, just like he would get if his file was bigger than the bucket setting.

The server (a console user) gets invalid input because his action is to update the value.

Does it make sense?

Reply:
not really, at the end of the day, $maximumFileSize is a parameter passed through the API and needs to use the `GENERAL_ARGUMENT_INVALID` error and not even  `ATTRIBUTE_VALUE_INVALID `

`ATTRIBUTE_VALUE_INVALID ` is meant to be used for database attributes . 
We're dealing with Utopia parameters here 😄 

Also refer to my comments above about using the `Range` Validator to do this check

Reply:
Okay. In the previous case, the parameter is passed through the API, here we fetch it from the Database 👍

Reply:
Error code makes sense, I would rewrite the error message a bit.

Reply:
No still, 400 series errors are for user errors and it may not be appropriate to use it in the context of the `createFile` endpoint where `$maximumFileSize` is **not passed** to it.   

I still think this is a server error to be treated with 5xx error codes. 

In the previous instances, where `maximumFileSize` is passed as a parameter to the endpoint, it makes sense to use a 4xx error code. 

So, I would stick to a 5xx code but the description looks like it could use some edits.

Reply:
This is now 500 server error, as suggested below.